prov/util/src/util_cq.c

```diff
@@ -203,21 +203,35 @@ ssize_t ofi_cq_readerr(struct fid_cq *cq_fid, struct fi_cq_err_entry *buf,
 	struct util_cq *cq;
 	struct util_cq_err_entry *err;
 	struct slist_entry *entry;
-	ssize_t ret;
+	char *err_buf_save = NULL;
+	ssize_t ret = -FI_EAGAIN;
 
 	cq = container_of(cq_fid, struct util_cq, cq_fid);
 	fastlock_acquire(&cq->cq_lock);
-	if (!ofi_cirque_isempty(cq->cirq) &&
-	    (ofi_cirque_head(cq->cirq)->flags & UTIL_FLAG_ERROR)) {
-		ofi_cirque_discard(cq->cirq);
-		entry = slist_remove_head(&cq->err_list);
-		err = container_of(entry, struct util_cq_err_entry, list_entry);
-		*buf = err->err_entry;
-		free(err);
-		ret = 0;
-	} else {
-		ret = -FI_EAGAIN;
+	if (ofi_cirque_isempty(cq->cirq) ||
+		!(ofi_cirque_head(cq->cirq)->flags & UTIL_FLAG_ERROR))
+		goto unlock;
+
+	ofi_cirque_discard(cq->cirq);
+	entry = slist_remove_head(&cq->err_list);
+	err = container_of(entry, struct util_cq_err_entry, list_entry);
+	if (buf->err_data_size) {
```

Comment:
We need to include a check for what version the application is attempting to use.  This check assumes the apps supports 1.5.